metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -492,9 +493,13 @@ class MetalsLanguageServer(
       statusBar,
       compilers
     )
+    scalafixProvider = ScalafixProvider
+      .from("2.12", buildTargets, buffers, workspace)
+      .get // TODO fail + scalafixBinary should be get from BuildTargets
```

Comment:
How to get the scala version here? 
Or should I have many scalafix instances downloaded (lazily) and cached (for each scalaVersion) in ScalafixProvider. 
Right now the only way I found the get the scalaVersion, is when I am formatting a file => 
```
      identifier <- buildTargets.inverseSources(file)
      scalaBuildTarget <- buildTargets.scalaInfo(identifier)
      scalaVersion = scalaBuildTarget.getScalaVersion
``` 
Another question: `ScalafixProvider.from(` is a method that can fail, if it's not able to fetch the artifact. Should it fail ? should the fetch be done here? We can also not fail, and in this case, we disable the organize import until the next restart?

Reply:
> How to get the scala version here?

That's exactly how you can get the version right now. We need the information from the build target. It also has `scalaBinaryVersion`, which will be useful here.
However, I think you should also have scalafix instances populated lazily for each scala version. Usually it will really be one per workspace.

As for `ScalafixProvider.from( ` we should handle the case it failed to download and show a message to the user in that case via `client.showMessage()`. We can retry downloading in case the user tries to organize imports again, no need to disable it as long as we send a clear message what is wrong.

Reply:
ok!